text/0101-revamping-the-sdk-performance-api.md

```diff
@@ -0,0 +1,268 @@
+| `data`            | Arbitrary additional data on a span, like `extra` on the top-level event. We maintain [conventions for span data keys and values](https://develop.sentry.dev/sdk/performance/span-data-conventions/). | Object          |
+
+As you can see, the fields on the transaction/span differ in a few ways, the most noteable of which is that transactions have `name` while spans have `description`. This means that spans and transactions are not interchangeable, and users have to know the difference between the two.
+
+In addition, user's have the burden to understand the differences between `name`/`description`/`operation`. `operation` in particular can be confusing, as it overlaps with transaction `name` and span `description`. In addition, `operation` is not a required field, which means that it is not clear what the default value should be.
+
+Transactions also have no mechanism for arbitrary additional data like spans do with `data`. Users can choose to add arbitrary data to transactions by adding it to the `contexts` field (as transactions extend the event schema), but this is not obvious and not exposed in every SDK. Since contexts are already well defined in their own way, there is no way of using [Sentry's semantic conventions for span data](https://develop.sentry.dev/sdk/performance/span-data-conventions/) for transactions.
+
+### New Span Schema
+
+To simplify how performance data is consumed and understood, we are proposing a new span schema that the SDKs send to Sentry. The new span schema aims to be a superset of the [OpenTelemetry span schema](https://github.com/open-telemetry/opentelemetry-proto/blob/4967b51b5cb29f725978362b9d413bae1dbb641c/opentelemetry/proto/trace/v1/trace.proto) and have a minimal top level API surface. This also means that spans can be easily converted to OpenTelemetry spans and vice versa.
+
+The new span schema is as follows:
+
+| Span Field        | Description                                                  | Type            | Notes                                                                                                                                                                                                                                                              |
+| ----------------- | ------------------------------------------------------------ | --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
+| `name`            | The name of the span                                         | String          | Should be low cardinality. Replacing span `description`                                                                                                                                                                                                            |
```

Comment:
I believe is important to keep `description`, how would I set the SQL statement of a DB query if it has to be low cardinality? One can say you can do it under the `data` field, so the `name` then would be similar to the former `operation`? some operations just don't have a unique name, is the grouping of transactions still per `name`?

Reply:
In the OpenTelemetry conventions, it's recommended db spans be named <db.operation> <db.name>.<db.sql.table> - like `SELECT bestDb.orders`, but the only thing they require is that something is low cardinality, so the db query also works here. In most cases, we can just add whatever we were setting in `span.description` to `span.name` (as long as it's low cardinality).

Grouping of transactions/spans would still be by name, but there is more freedom about what that should be. The product instead relies on span attributes to make this work.

Reply:
I see, DB queries are high cardinality though, at least for bigger databases.
Most of the time interceptors/hooks just give you a `String`, so to build `SELECT bestDb.orders` requires parsing the String on the client which is also not ideal.
Alright, this is more an implementation detail anyway, thanks.

Reply:
I'm OK removing `op` and adding a new low cardinality format of OpenTelemetry as you described @AbhiPrasad, but I think the description we currently have for db queries, HTTP requests, and such is highly useful. Where would that end up now?

Reply:
@philipphofmann the span attributes, aka `span.data`